src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -512,9 +512,25 @@ export class ConnectionDialogService implements IConnectionDialogService {
 				}, 10);
 				return;
 			}));
-
 		}
+
 		this._logService.error(message);
-		this._errorMessageService.showDialog(severity, headerTitle, message, messageDetails, actions);
+
+		// Set instructionText for MSSQL Provider Encryption error code -2146893019 thrown by SqlClient when certificate validation fails.
+		let instructionText: string;
+		if (errorCode === -2146893019) {
+			let enableTrustServerCert = localize('enableTrustServerCertificate', "Enable Trust server certificate");
+			instructionText = localize('trustServerCertInstructionText', `Encryption was enabled on this connection, review your SSL and certificate configuration for the target SQL Server, or enable 'Trust server certificate' in the connection dialog.
+
+			Do you want to enable 'Trust server certificate' on this connection and retry?`);
+			actions.push(new Action('trustServerCert', enableTrustServerCert, undefined, true, async () => {
+				this._model.options['trustServerCertificate'] = true;
```

Comment:
I suggest starting a const enum for well-known connection properties like this so we can easily re-use them and ensure they're consistent.